packages/ui-components/src/components/range/range.tsx

```diff
@@ -0,0 +1,69 @@
+import { Component, Host, h, Prop, Element, EventEmitter, Event } from '@stencil/core';
+import { IRange, IRangeEvents } from './range.types';
+import { getInputPercentageFromValue, getOffset } from './range.helper';
+
+@Component({
+	tag: 'kv-range',
+	styleUrl: 'range.scss',
+	shadow: true
+})
+export class KvRange implements IRange, IRangeEvents {
+	@Element() el!: HTMLKvRangeElement;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) min: number;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) max: number;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) value?: number = 0;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) step?: number = 1;
+	/** @inheritdoc */
+	@Event() valueChange: EventEmitter<number>;
+
+	componentDidRender() {
+		const slider = this.el.shadowRoot.getElementById('slider') as HTMLElement;
+		this.applyCssStyles();
+		slider.oninput = () => {
+			this.applyCssStyles();
+		};
+	}
+
+	private applyCssStyles = () => {
+		const rangeInputValue = this.el.shadowRoot.querySelector('input');
+
+		const selector = this.el.shadowRoot.getElementById('selector');
+
+		let inputValue = rangeInputValue.value;
+		const percentage = getInputPercentageFromValue(inputValue, this.min, this.max);
+		const offSet = getOffset(percentage);
+
+		selector.style.left = percentage + '%';
+		selector.style.marginLeft = offSet + 'px';
+		rangeInputValue.style.background = `linear-gradient(90deg, #005cc7 ${percentage}%, #121212 ${percentage + 0.1}%)`;
```

Comment:
You can't have hardcoded colors like you have here. 
Suggestion: 
`rangeInputValue.style.background = linear-gradient(90deg, var(--slider-background-filled) ${percentage}%, var(--slider-background-empty) ${percentage + 0.1}%);`
and  add these var in scss file